libs/utils/utils.js

```diff
@@ -1253,7 +1242,6 @@ async function processSection(section, config, isDoc) {
   preloadBlockResources(section.preloadLinks);
   await Promise.all([
     decoratePlaceholders(section.el, config),
```

Comment:
This still doesn't look right, we seen good effects on LCP scores by parallelizing the loading efforts. You want to structure the code in a way that ideally we load all of those at the same time, that we load the blocks.
- Icon.js
- Icon.css
- federal.js
- Actual icons needed for LCP only

Before: 
![Screenshot 2024-10-22 at 10 11 59](https://github.com/user-attachments/assets/175ab7bd-a67d-4196-a646-2e53c3c802cc)

After: https://main--cc--adobecom.hlx.live/de/products/illustrator/features?milolibs=rparrish-fed-icons--milo--adobecom
![Screenshot 2024-10-22 at 10 11 03](https://github.com/user-attachments/assets/65ddc5d9-40fa-4191-bd66-5a2393806510)

Reply:
@mokimo I think `icon.js` needs to wait for `federated.js` and then make the network call to fetch the `svg` icons. The older structure was that all the logic was within `icon.js` along with the `svg` icons. Let me know if I am wrong.

Reply:
The idea would be to structure the code a bit like this again:
```
async function decorateIcons(area, config) {
  // see if there is an icon within the LCP section
  const icons = area.querySelectorAll('span.icon');
  if (icons.length === 0) return;

  // if we have an icon, we want to load icon.js+federated+js+icon.css+the icons relevant for the LCP
  const { base } = config;
  loadStyle(`${base}/features/icons/icons.css`);
  loadLink(`${base}/img/icons/federal.js`, { rel: 'preload', as: 'script', crossorigin: 'anonymous' });
  // load or preload the icon rather than having to wait for the icons.js to discover which icon is actually needed
  const { default: loadIcons } = await import('../features/icons/icons.js');
  await loadIcons(icons, config);
}
```

Reply:
We'll want to hint what to load to the browser *before* waiting for the blocking chains, so that the browser can load everything ASAP rather than when discovered sometime one after eachother

Reply:
@mokimo  The current code structure moves all this logic to `icon.js` file. I think, we can move back the loading of files `federated.js` and `icons.css` to load in parallel. But for the actual icon itself, it must wait for the `federated.js` to get the federal url and make the network call, is it not? 
<img width="882" alt="Screenshot 2024-10-24 at 1 15 41 PM" src="https://github.com/user-attachments/assets/c130f3c7-986e-4396-9b38-7cf2f1315507">

Reply:
@saugatmalla that's a great point that we'd need to load federal to determine which icon to load.

Option 1: Take a small performance and keep icon loading logic within icon.js 
Option 2: Add federal to utils.js and some logic to preload/load the first icon, which should be slightly better from a performance standpoint.

I would tend to say Option-2 is slightly favorable, as we have more and more features relying on this centralized-content feature and thus its justified within `utils` directly. @overmyheadandbody @narcis-radu @robert-bogos any opinions that?

Reply:
@mokimo here's my take: 
I'm pretty neutral to adding the federal stuff to utils.js. Maybe just ever so slightly against, because I'm not sure it's necessary for LCP, but I don't think adding it would have a strong negative impact either. 

Personally though, I think the real issue here is that the request to get icons is blocking. I think a better approach would be to load the icons like we're doing, but non-blocking, and reserve space for the icons until they've loaded in to prevent any CLS. 
This way everything else still loads quickly and we're not needlessly loading icons.svg unless there are actual icons from there that need to be displayed.

Reply:
We did have a draft PR for [loading icons post LCP](https://github.com/orgs/adobecom/discussions/2325) - but ran into some more complex CLS issues according to @ryanmparrish ... if we can make it happen without the CLS tradeoff, I'd love to have it.

Reply:
>  I think, we can move back the loading of files federated.js and icons.css to load in parallel. 

This is the most un-intrusive change for now, so let's go with this to unblock this. @saugatmalla would you mind making the changes as long as Ryan is out?

Please also ensure that we only load federal.js/icon.js if there are icons within the first LCP section and that we load icons on a section-to-section basis. (e.g. we don't need to load icons somewhere on the bottom of the page that aren't strictly needed for the LCP, if we are just about to load the first section.)

Reply:
I noticed that icons are rather being fetched after sections are processed, so that it doesn't block for LCP ([see this feedback](#issuecomment-2394761870)). So `icon.js`, `icons.css` and `federared.js` can just be loaded in the later stage. 

I moved the blocking code piece of `icon.js` to `utils.js`.